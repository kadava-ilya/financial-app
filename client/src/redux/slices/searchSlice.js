import {createSlice} from "@reduxjs/toolkit";

const initialState = {
    inputText: '',
}

const searchSlice = createSlice({
    name: 'inputValue',
    initialState,
    reducers: {
        setInputValue(state, action) {
            state.inputText = action.payload;
        },
        setInputClear(state) {
            state.inputText = ''
        }
    }
})

export const {setInputValue, setInputClear} = searchSlice.actions;

export default searchSlice.reducer;