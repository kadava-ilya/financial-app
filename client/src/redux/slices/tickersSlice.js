import {createSlice} from "@reduxjs/toolkit";

const initialState = {
    tickers: [],
    removedTicker: [],
    timeRange: 5000,
}

const tickersSlice = createSlice({
    name: 'tickers',
    initialState,
    reducers: {
        setTickers(state, action) {
            state.tickers = action.payload;
        },
        removeTicker(state, action) {
            state.removedTicker.push(action.payload);
        },
        showTicker(state, action) {
            state.removedTicker = state.removedTicker.filter(ticker => {return ticker !== action.payload})
        },
        setTimeRange(state, action) {
            state.timeRange = action.payload;
        }
    }
})


export const {setTickers, removeTicker, showTicker, setTimeRange} = tickersSlice.actions;

export default tickersSlice.reducer;