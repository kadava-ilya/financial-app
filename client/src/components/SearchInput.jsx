import React from "react";
import {Box, InputAdornment, TextField} from "@mui/material";
import {setInputValue, setInputClear} from "../redux/slices/searchSlice";
import {useDispatch, useSelector} from "react-redux";
import {ClearButton} from "./ClearButton";
import {selectInputValue} from "../redux/selectors";

export const SearchInput = () => {

    const dispatch = useDispatch();
    const inputValue = useSelector(selectInputValue);

    return (
        <Box component="form">
            <TextField
                id=''
                sx={{minWidth: 300}}
                value={inputValue}
                placeholder='Search'
                onChange={(e) => dispatch(setInputValue(e.target.value))}
                InputProps={{
                    endAdornment: (
                        <InputAdornment position="end">
                            {inputValue && (
                                <ClearButton onClick={setInputClear()}/>
                            )}
                        </InputAdornment>
                    ),
                }}
            />
        </Box>
    )
}