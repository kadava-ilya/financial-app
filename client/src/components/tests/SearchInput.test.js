import {SearchInput} from "../SearchInput";
import {render, screen, fireEvent} from "@testing-library/react";
import * as reduxHooks from 'react-redux'
import * as actions from '../../redux/slices/searchSlice'
import userEvent from "@testing-library/user-event";

jest.mock('react-redux')

const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch')

describe('SearchInput', () => {
    it('should create SearchInput', () => {
        mockedDispatch.mockReturnValue(jest.fn());

        const component = render(<SearchInput />)

        expect(component).toMatchSnapshot()
    })

    it('SearchInput renders', () => {
        mockedDispatch.mockReturnValue(jest.fn());

        render(<SearchInput />)

        expect(screen.getByPlaceholderText('Search')).toBeInTheDocument();
    })

    it('should dispatch setInputValue on input value', () => {
        const mockedSetInputValue = jest.spyOn(actions, 'setInputValue')

        render(<SearchInput />)

        userEvent.type(screen.getByPlaceholderText(/Search/), 'Test')

        expect(mockedSetInputValue).toHaveBeenCalledTimes(4)
    })
})