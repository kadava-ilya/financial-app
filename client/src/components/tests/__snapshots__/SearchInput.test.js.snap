// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should create SearchInput 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-ysnqqo-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id=""
              placeholder="Search"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="MuiBox-root css-0"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-ysnqqo-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id=""
            placeholder="Search"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
